introduction.md

```diff
@@ -0,0 +1,144 @@
+
+- The Unvault Emergency TX would spend the Unvault TX to pay to the `emergency output descriptor`.
+- The Cancel TX would spend the Unvault TX to pay back to the `deposit output descriptor`.
+
+This way, funds are protected _before_ delegation to fund managers. Finally, the Unvault TX is signed. It spends the Deposit TX and pays to either:
+
+- the managers' K-of-M multi-signature + J cosigning servers + time-lock of X blocks, or 
+- to the stakeholders' N-of-N multi-signature without delay. 
+
+Managers can attempt to use the Spend TX to pay external wallets, but are forced to wait for the Unvault time-lock to expire before their transaction finalizes. During that delay period, a Cancel or Unvault Emergency TX can be broadcasted if the manager is breaching pre-defined policy. 
+
+Please check the complete specification of [transactions](transactions.md) for more details.
+
+## Infrastructure and Feature Choice
+
+Users can choose features to suit their needs. Different features require different infrastructure to operate. 
+
+First, there are base requirements common to all Revault deployments. Each stakeholder and manager controls a wallet and a signing device. An untrusted Coordinator server is used to route messages and thus enable wallets to update and synchronise their state.  
+
+- **Wallet**: an open-source software running with a bitcoind back-end used to keep track of the users' co-owned coins, transaction signatures, and to connect with a signing device and infrastructure servers.
+
+- **Signing Device**: a hardware module used to store private keys, public keys and wallet descriptors and to handle signing operations. As user verification is an important part of the Revault security model, such a device must allow the user to check the message integrity before signing it. 
+
+- **Coordinator server**: An always-online server used for efficient message routing among participants and other servers. Its primary use is for exchanging signatures and Spend TXs. 
+
+With these in place, stakeholders are able to delegate bitcoin to fund managers and manually broadcast Cancel, Emergency and Unvault-Emergency TXs. Fund managers will also be able to manually broadcast Cancel TXs to recover from mistakes. 
+
+Next, users who want additional features will require additional infrastructure components. Each stakeholder may control one or more watchtowers and/ or a co-signing server.
+
+- **Watchtower**: An always-online server that monitors the bitcoin network and chain, as well as polls the coordinator for Spend TXs to perform its responsibilities (automated unvault policy enforcement, automated spend policy enforcement, emergency deterrent). 
+
+- **Co-signing Server**: An always-online server that acts as a 'stupidly simple' anti-replay signing oracle: it will accept to sign any Spend TX, *but only once*. This is necessary to stop the managers from changing the Spend TX destination after the Unvault TX’s time-lock has expired. 
```

Comment:
> from changing the Spend TX destination after the Unvault TX’s time-lock has expired.

Huh, no. After timelock expiration if the Unvault was rogue but not revaulted, it's game over.

```suggestion
- **Co-signing Server**: An always-online server that acts as a 'stupidly simple' anti-replay signing oracle: it will accept to sign any Spend TX, *but only once*. This is necessary to stop the managers from malleating the Spend transaction after the Unvault transaction(s) were broadcast. 
```